services/motion/builtin/builtin_test.go

```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		req.Obstacles = []spatialmath.Geometry{obstacleLeft}
+
+		// construct move request
+		planExecutor, err := ms.(*builtIn).newMoveOnMapRequest(ctx, req, nil, 0)
+		test.That(t, err, test.ShouldBeNil)
+		mr, ok := planExecutor.(*moveRequest)
+		test.That(t, ok, test.ShouldBeTrue)
+
+		// construct plan
+		planResp, err := mr.Plan(ctx)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(planResp.Waypoints), test.ShouldBeGreaterThan, 2)
+
+		// place obstacle in opposte position and show that the generate path
+		// collides with obstacleRight
+		obstacleRight, err := spatialmath.NewBox(
+			spatialmath.NewPose(r3.Vector{0.89627e3, -0.37192e3, 0},
+				&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: -45}),
+			r3.Vector{900, 10, 10},
+			"obstacleLeft",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		wrldSt, err := referenceframe.NewWorldState(
+			[]*referenceframe.GeometriesInFrame{
+				referenceframe.NewGeometriesInFrame(
+					referenceframe.World,
+					[]spatialmath.Geometry{obstacleRight},
+				),
+			}, nil,
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = motionplan.CheckPlan(
+			mr.planRequest.Frame,
+			planResp.Motionplan,
+			wrldSt,
+			mr.planRequest.FrameSystem,
+			spatialmath.NewPose(
+				r3.Vector{X: 0.58772e3, Y: -0.80826e3, Z: 0},
+				&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 0},
+			),
+			referenceframe.FloatsToInputs([]float64{0, 0, 0}),
+			spatialmath.NewZeroPose(),
+			lookAheadDistanceMM,
+			logger,
+		)
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("fail due to obstacles enclosing goals", func(t *testing.T) {
+		// define static obstacles
+		obstacleTop, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.64603e3, 0.77151e3, 0}),
+			r3.Vector{400, 10, 10},
+			"obstacleTop",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleBottom, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.64603e3, 0.42479e3, 0}),
+			r3.Vector{400, 10, 10},
+			"obstacleBottom",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleLeft, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.47525e3, 0.65091e3, 0}),
+			r3.Vector{10, 400, 10},
+			"obstacleLeft",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleRight, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.82183e3, 0.64589e3, 0}),
+			r3.Vector{10, 400, 10},
+			"obstacleRight",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		req.Obstacles = []spatialmath.Geometry{obstacleTop, obstacleBottom, obstacleLeft, obstacleRight}
+
+		// construct move request
+		planExecutor, err := ms.(*builtIn).newMoveOnMapRequest(ctx, req, nil, 0)
+		test.That(t, err, test.ShouldBeNil)
+		mr, ok := planExecutor.(*moveRequest)
+		test.That(t, ok, test.ShouldBeTrue)
+
+		// construct plan
+		planResp, err := mr.Plan(ctx)
+		test.That(t, err, test.ShouldBeError, errors.New("context deadline exceeded"))
```

Comment:
How long does this take to run? I don't see a timeout on the context

Reply:
```
	extra := map[string]interface{}{
		"motion_profile": "position_only",
		"timeout":        15.,
		"smooth_iter":    10.,
	}
```
we set a timeout through extras

Reply:
15 seems kind of high, can we get away with something lower?

Reply:
Is 5 too aggressive?

Reply:
i think 5 would work. the success case is a solving for path for a trivial goal - which takes `0.4` seconds on my laptop.